dataStructures/uniqueItems/solutionA/src/main/java/oointro/DungeonLoot.java

```diff
@@ -0,0 +1,37 @@
+package oointro;
+import java.util.Collections;
+import java.util.Comparator;
+import java.util.HashMap;
+import java.util.HashSet;
+import java.util.List;
+import java.util.ArrayList;
+
+class LootComparator implements Comparator<Loot> {
+    @Override
+    public int compare(Loot l1, Loot l2) {
+        return l1.getName().compareTo(l2.getName());
+    }
+}
+
+public class DungeonLoot {
+    HashSet<Loot> storeLoot = new HashSet<>();
+
+    //methods
+    public void addLoot(Loot newLoot){
+        storeLoot.add(newLoot);
+    }
+
+    public void removeLoot(String name){
+        storeLoot.removeIf(loot -> loot.getName() == name);
+    }
```

Comment:
that is a really good use of a lambda expression.  Its going to confuse a lot of  people though.   Could you write a non-lambda version of the method and leave it commented out just below this one with a bit of commentary about the  two?

Reply:
Ok good call 👍